plugin/pulpcore/plugin/changeset/model.py

```diff
@@ -147,41 +180,47 @@ class RemoteArtifact(Remote):
 
     Attributes:
         download (pulpcore.download.Download): An object used to download the content.
-        content (RemoteContent): The associated remote content.
+        content_artifact (ContentArtifact): The association between an Artifact and Content.
         path (str): Absolute path to the downloaded file.  May be (None) when
             downloading is deferred or the artifact has already been downloaded.
 
     Examples:
         >>>
-        >>> from pulpcore.plugin.models import Artifact
+        >>> from pulpcore.plugin.models import ContentArtifact, DeferredArtifact
         >>>
-        >>> model = Artifact(...)  # DB model instance.
+        >>> model = DeferredArtifact(...)
+        >>> content_artifact = ContentArtifact(...)
         >>> download = ...
         >>> ...
-        >>> artifact = RemoteArtifact(model, download)
+        >>> artifact = RemoteArtifact(model, content_artifact, download)
         >>>
     """
 
     __slots__ = (
         'content',
+        'content_artifact',
```

Comment:
Let's make this `relative_path` and manage the `ContentArtifact` for the plugin writer.